test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs

```diff
@@ -320,6 +322,36 @@ public void AutoFitWithPresplittedData()
 
         }
 
+        [Fact]
+        public void AutoFitMaxExperimentTimeTest()
+		{
+            // 1 Binary classification experiment takes less than 5 seconds.
+            // System.OperationCanceledException is thrown when ongoing experiment
+            // is canceled and at least one model has been generated.
+            var context = new MLContext(1);
+            var dataPath = DatasetUtil.GetUciAdultDataset();
+            var columnInference = context.Auto().InferColumns(dataPath, DatasetUtil.UciAdultLabel);
+            var textLoader = context.Data.CreateTextLoader(columnInference.TextLoaderOptions);
+            var trainData = textLoader.Load(dataPath);
+            var experiment = context.Auto()
+                .CreateBinaryClassificationExperiment(5)
+                .Execute(trainData, new ColumnInformation() { LabelColumnName = DatasetUtil.UciAdultLabel });
+            RunDetail<BinaryClassificationMetrics>[] runDetails = experiment.RunDetails.Where(r => r.Model != null).ToArray();
+            foreach(RunDetail<BinaryClassificationMetrics> runDetail in runDetails)
+			{
+                ModelContainer modelContainer = GetInstanceField(typeof(RunDetail<BinaryClassificationMetrics>), runDetail, "_modelContainer") as ModelContainer;
+                MLContext thisContext = GetInstanceField(typeof(ModelContainer), modelContainer, "_mlContext") as MLContext;
+                Assert.True((thisContext.Model.GetEnvironment() as ICancelable).IsCanceled);
```

Comment:
Shouldn't just the last model be canceled, the one model which was training when the timer expired?

Side note: there's mixed tab/spaces for indentation in the new code.

Reply:
As we're now adding new MLContext's per each model run in addition to the MLContext which is tied to the origination of the experiment, I think all additional MLContexts should be canceled. Should this be so, or only the MLContext that is tied to the last model be canceled?

Reply:
That is so, I have now made it so that the last model that was in the middle of training is canceled.

Reply:
The goal of the refitting unit test I was recommending is to ensure the refitting workflow continues working after this PR's change  -- https://github.com/dotnet/machinelearning/pull/5445#discussion_r510599919.

Method 1:
One way to have the unit test pass is to only cancel the last MLContext (more specifically, anything that was actually canceled). Then the `experimentResult.BestRun` (and other non-canceled iterations) can be refit.

The multi-step reasoning is, if we cancel a MLContext it is no longer usable for training. The MLContext is stored as part of the Estimator returned from the AutoML API call. The Estimator is used to retrain the pipeline found by AutoML on additional data, for instance thrice training on (train/valid/test) data to make a more useful model, which is the [recommend final step](https://github.com/dotnet/machinelearning-samples/blob/ab652eadb1f7552c1816bc4d2ed0a27ef4675a37/samples/csharp/getting-started/AdvancedExperiment_AutoML/AdvancedTaxiFarePrediction/Program.cs#L186-L197) in the AutoML workflow.

If we cancel all MLContexts, retraining will only be doable if the time limit isn't hit. Sometimes the user will have a canceled context, sometimes not. 

Method 2:
Another possible way is if we keep around a second copy of the Estimator, and return it to the user. We fit one to make the model and cancel it after AutoML uses it, and returns a clean non-canceled copy of the Estimator to the user.